cmake/FindCharm.cmake

```diff
@@ -91,22 +91,33 @@ execute_process(
   COMMAND bash -c "${CHARM_BUILDING_BLOCKS_CMD}"
   OUTPUT_VARIABLE CHARM_BUILDING_BLOCKS
   ERROR_VARIABLE CHARM_BUILDING_BLOCKS_ERROR
-  ECHO_ERROR_VARIABLE
   OUTPUT_STRIP_TRAILING_WHITESPACE
   )
 if(NOT CHARM_BUILDING_BLOCKS)
   if(CHARM_BUILDING_BLOCKS_ERROR MATCHES "lib_so directory not found"
       AND CHARM_SHARED_LIBS)
-    message(FATAL_ERROR "You are requesting a shared-library build but it "
-      "looks like Charm++ is not built with shared-library support. Build "
-      "Charm++ with the '--build-shared' option. Alternatives:\n"
-      "  - Link Charm++ statically (-D CHARM_SHARED_LIBS=OFF). This requires "
-      "Charm++ was built with '-fPIC'.\n"
-      "  - Build SpECTRE statically (-D BUILD_SHARED_LIBS=OFF).")
+    message(STATUS "Requested Charm++ shared libraries, but it looks like "
```

Comment:
All other packages don't show this kind of info. It might be useful instead to print a status message like `"Using Charm++ static libraries. Please make sure Charm++ was built with '-fPIC'"` instead

Reply:
- There's no way to check in cmake if you have manually built charm with `-fPIC` or not, hence the warning above. It says "make sure", not "you haven't".
- It's way easier to always build charm with `--build-shared`. Then it builds static libs with `-fPIC` and the shared libs are also available. Also the `-fPIC` shows up in cmake, so you don't get the warning.
- I think it's important to tell users when the cmake script decides to do something different than what they requested. Hence the status message in line 99. I'd even prefer to drop this hunk and have the user explicitly set `CHARM_SHARED_LIBS`. At least they know what's going on then 🤷‍♂️

Reply:
The point is that on supercomputers you shouldn't build with shared libs, and I don't think it's reasonable for us to effectively force people to use giant CMake commands and bombard them with incorrect warnings just for using a valid configuration. Having a status printed out if Charm++ is built with static libs that they need to be built with `-fPIC` is fine, but the current approach is extremely tedious and verbose.

Reply:
if you aren't building with shared libs then all is fine anyway :) You're not getting warnings about `-fPIC` or anything else.

Reply:
This is resolved in "fixup Fall back to static libs" now I think.